### PR TITLE
Add GitPod to Marquez

### PR DESCRIPTION
**Pull request by phixMe**
This adds [gitpod.io](https://gitpod.io) to Marquez along with an associated configuration file. 

The docker configuration for the api project was not working due to subsequent `COPY` instructions resulting in a noop and failing the multi-staged build. Thus, I added some separator instructions to prevent this issue from occurring. More information can be found [here.](https://stackoverflow.com/questions/51115856/docker-failed-to-export-image-failed-to-create-image-failed-to-get-layer)

This button has been added to the root readme to allow anybody to create a on demand ephemeral developer environment.
[Open in Gitpod](https://gitpod.io/#https://github.com/MarquezProject/marquez/tree/feature/gitpod)

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [x] You've updated the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) with details about your change under the "Unreleased" section (_if relevant, depending on the change, this may not be necessary_)
- [x] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 Marquez is an open source **metadata service** for the **collection**, **aggregation**, and **visualization** of a data ecosystem's metadata. It maintains the provenance of how datasets are consumed and produced, provides global visibility into job runtime and frequency of dataset access, centralization of dataset lifecycle management, and much more. Marquez was released and open sourced by [WeWork](https://www.wework.com).
 
+## Quick Demo
+[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/MarquezProject/marquez/tree/feature/gitpod)
```
Comment (phixMe) [inline README.md:11]: This needs to be changed to not reference the branch before merging.

File: Dockerfile [hunk 2]
```diff
@@ -9,8 +9,11 @@ RUN ./gradlew --version
 FROM base AS build
 WORKDIR /usr/src/app
 COPY build.gradle build.gradle
+RUN true
```
Comment (wslulciuc) [inline Dockerfile:12]: Each `RUN` creates a new image layer. So, I'm not a huge fan of this workaround. Looks like another option is to set `DOCKER_BUILDKIT=1` when building the docker image. Mind trying

```
DOCKER_BUILDKIT=1 ./docker/up.sh --seed
```

in the `.gitpod.yml`?

File: README.md (hunk 1 above)
Comment (wslulciuc) [reply README.md:11]: I'd move this under an "Try it!" section above "Quickstart":

```
## Status
...

## Try it!

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/MarquezProject/marquez/tree/feature/gitpod)

## Quickstart
...
```

Comment (phixMe) [reply README.md]: Sounds good.

**Review by wslulciuc — APPROVED** [review]
❤️

**Review by julienledem — APPROVED** [review]
This is amazing

**Pull request merged**